OpenEdXMobile/src/main/java/org/edx/mobile/repositorie/CourseDatesRepository.kt

```diff
@@ -31,19 +34,27 @@ class CourseDatesRepository {
     /**
      * Fetch course dates against course Id.
      */
-    fun getCourseDates(courseId: String, callback: NetworkResponseCallback<CourseDates>) {
-        courseAPI.getCourseDates(courseId).enqueue(object : Callback<CourseDates> {
-            override fun onResponse(call: Call<CourseDates>, response: Response<CourseDates>) {
-                callback.onSuccess(Result.Success<CourseDates>(isSuccessful = response.isSuccessful,
-                        data = response.body(),
-                        code = response.code(),
-                        message = response.message()))
-            }
+    fun getCourseDates(courseId: String, forceRefresh: Boolean, callback: NetworkResponseCallback<CourseDates>) {
+        if (forceRefresh || courseDates == null) {
+            courseAPI.getCourseDates(courseId).enqueue(object : Callback<CourseDates> {
```

Comment:
please add `courseDates = null`, before making the service call, cuz if the user opens the second course, app shows the dates of previously opened course dates.